File: app/src/main/java/org/mifos/mobile/presenters/NotificationPresenter.java
package org.mifos.mobile.presenters;

import android.content.Context;

import org.mifos.mobile.R;
import org.mifos.mobile.api.DataManager;
import org.mifos.mobile.injection.ActivityContext;
import org.mifos.mobile.models.notification.MifosNotification;
import org.mifos.mobile.presenters.base.BasePresenter;
import org.mifos.mobile.ui.views.NotificationView;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.observers.DisposableObserver;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by dilpreet on 14/9/17.
 */

public class NotificationPresenter extends BasePresenter<NotificationView> {

    private DataManager manager;
    private CompositeDisposable compositeDisposable;

    /**
     * Initialises the LoginPresenter by automatically injecting an instance of
     * {@link DataManager} and {@link Context}.
     *
     * @param manager DataManager class that provides access to the data
     *                via the API.
     * @param context Context of the view attached to the presenter. In this case
     *                it is that of an {@link androidx.appcompat.app.AppCompatActivity}
     */
    @Inject
    public NotificationPresenter(DataManager manager, @ActivityContext Context context) {
        super(context);
        this.manager = manager;
        compositeDisposable = new CompositeDisposable();
    }

    @Override
    public void attachView(NotificationView mvpView) {
        super.attachView(mvpView);
    }

    @Override
    public void detachView() {
        super.detachView();
        compositeDisposable.clear();
    }

    public void loadNotifications() {

        checkViewAttached();
        getMvpView().showProgress();
        compositeDisposable.add(manager.getNotifications()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribeWith(new DisposableObserver<List<MifosNotification>>() {
                    @Override
                    public void onComplete() {

                    }

                    @Override
                    public void onError(Throwable e) {
                        getMvpView().hideProgress();
                        getMvpView().showError(context
                                .getString(R.string.notification));
                    }

                    @Override
                    public void onNext(List<MifosNotification> notificationModels) {
                        getMvpView().hideProgress();
                        getMvpView().showNotifications(notificationModels);
                    }
                }));
    }
}

File: app/src/main/java/org/mifos/mobile/presenters/NotificationPresenter.kt
package org.mifos.mobile.presenters

import android.content.Context

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.observers.DisposableObserver
import io.reactivex.schedulers.Schedulers

import org.mifos.mobile.R
import org.mifos.mobile.api.DataManager
import org.mifos.mobile.injection.ActivityContext
import org.mifos.mobile.models.notification.MifosNotification
import org.mifos.mobile.presenters.base.BasePresenter
import org.mifos.mobile.ui.views.NotificationView

import javax.inject.Inject

/**
 * Created by dilpreet on 14/9/17.
 */
class NotificationPresenter @Inject constructor(private val manager: DataManager, @ActivityContext context: Context) :
        BasePresenter<NotificationView?>(context) {

    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
    override fun attachView(mvpView: NotificationView?) {
        super.attachView(mvpView)
    }

    override fun detachView() {
        super.detachView()
        compositeDisposable.clear()
    }

    fun loadNotifications() {
        checkViewAttached()
        mvpView?.showProgress()
        compositeDisposable.add(manager.notifications
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribeWith(object : DisposableObserver<List<MifosNotification?>?>() {
                    override fun onComplete() {}
                    override fun onError(e: Throwable) {
                        mvpView?.hideProgress()
                        mvpView?.showError(context
                                ?.getString(R.string.notification))
                    }

                    override fun onNext(notificationModels: List<MifosNotification?>) {
                        mvpView?.hideProgress()
                        mvpView?.showNotifications(notificationModels)
                    }
                }))
    }

}